src/include/execution/exec/output.h

```diff
@@ -45,23 +45,50 @@ class EXPORT OutputBuffer {
    */
   explicit OutputBuffer(sql::MemoryPool *memory_pool, uint16_t num_cols, uint32_t tuple_size, OutputCallback callback)
       : memory_pool_(memory_pool),
-        num_tuples_(0),
         tuple_size_(tuple_size),
-        tuples_(
-            reinterpret_cast<byte *>(memory_pool->AllocateAligned(BATCH_SIZE * tuple_size, alignof(uint64_t), true))),
-        callback_(std::move(callback)) {}
+        id_(0),
+        callback_(std::move(callback)) {
+          max_thread_ = std::thread::hardware_concurrency();
+          if (max_thread_ == 0) {
```

Comment:
I noticed that you did this in a few places. `std::thread::hardware_concurrency()` returns `0` only "If this value is not computable or well defined" whatever that means. I googled around and couldn't find any info on what would cause it to return 0. 🤷

Reply:
Yea I'm not sure in what cases this function would return 0, but it wouldn't hurt to be extra safe.